### PR TITLE
Improve sites data export

### PR DESCRIPTION
**Pull request by TobyRet**
## Context

In order for this export to be useful, we need to be able to filter out sites which are no longer available. 

## Changes proposed in this pull request

Made it so! 🖖 

## Link to Trello card

https://trello.com/c/zRMqdpyq/2710-dev-improve-the-query-we-use-for-the-sites-data-export

## Things to check

- [x] This code does not rely on migrations in the same Pull Request
- [ ] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [ ] API release notes have been updated if necessary
- [ ] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-teacher-training#azure-hosting-devops-pipeline)

### CODE REVIEW

File: app/models/site.rb [hunk 1]
```diff
@@ -1,5 +1,6 @@
 class Site < ApplicationRecord
   belongs_to :provider
+  has_one :course_option
```
Comment (stevehook) [inline app/models/site.rb:3]: Isn't this a `has_many` association?

File: app/services/support_interface/sites_export.rb [hunk 2]
```diff
@@ -18,7 +18,10 @@ def sites
   private
 
     def relevant_sites
-      Site.includes(:provider).where(providers: { sync_courses: true }).order('providers.code ASC')
+      Site.includes(:course_option, :provider)
```
Comment (stevehook) [inline app/services/support_interface/sites_export.rb:21]: `includes` will generate a `LEFT OUTER JOIN` in this case but I think you need an `INNER JOIN` so I think you might need a `joins(:course_options, :provider)` if I understand correctly (a big 'if').

Comment (TobyRet) [reply app/services/support_interface/sites_export.rb:21]: Yeah you're right - will make the change 👍

File: app/models/site.rb (hunk 1 above)
Comment (TobyRet) [reply app/models/site.rb:3]: Yep - I'll make the change

File: app/services/support_interface/sites_export.rb (hunk 2 above)
Comment (TobyRet) [reply app/services/support_interface/sites_export.rb:21]:
```
[5] pry(main)> Site.joins(:course_options, :provider).where(course_options: { site_still_valid: true }).where(providers: { sync_courses: true }).order('providers.code ASC')
  Site Load (5.2ms)  SELECT "sites".* FROM "sites" INNER JOIN "course_options" ON "course_options"."site_id" = "sites"."id" INNER JOIN "providers" ON "providers"."id" = "sites"."provider_id" WHERE "course_options"."site_still_valid" = $1 AND "providers"."sync_courses" = $2 ORDER BY providers.code ASC  [["site_still_valid", true], ["sync_courses", true]]
  ```

**Review by davidgisbey — APPROVED** [review]

**Pull request merged**